client/my-sites/design-preview/index.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import noop from 'lodash/noop';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import WebPreview from 'components/web-preview';
+import * as PreviewActions from 'state/preview/actions';
+import accept from 'lib/accept';
+import { updatePreviewWithChanges } from 'lib/design-preview';
+import layoutFocus from 'lib/layout-focus';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+const debug = debugFactory( 'calypso:design-preview' );
+
+const DesignPreview = React.createClass( {
+
+	propTypes: {
+		// Any additional classNames to set on this wrapper
+		className: React.PropTypes.string,
+		// True to show the preview; same as WebPreview.
+		showPreview: React.PropTypes.bool,
+		// The viewport device to show initially; same as WebPreview but defaults to 'tablet'.
+		defaultViewportDevice: React.PropTypes.string,
+		// Show close button; same as WebPreview.
+		showClose: React.PropTypes.bool,
+		// Elements to render on the right side of the toolbar; same as WebPreview.
+		children: React.PropTypes.node,
+		// A function to run when the preview has loaded. Will be passed a ref to the iframe document object.
+		onLoad: React.PropTypes.func,
+		// These are all provided by the state
+		previewMarkup: React.PropTypes.string,
+		customizations: React.PropTypes.object,
+		actions: React.PropTypes.object,
+		isUnsaved: React.PropTypes.bool,
+		selectedSiteId: React.PropTypes.number,
+	},
+
+	getDefaultProps() {
+		return {
+			showPreview: false,
+			defaultViewportDevice: 'tablet',
+			showClose: true,
+			customizations: {},
+			actions: {},
+			isUnsaved: false,
+			onLoad: noop,
+		};
+	},
+
+	componentDidMount() {
+		this.loadPreviewWithProps();
+	},
+
+	componentDidUpdate( prevProps ) {
+		// If there is no markup, fetch it
+		if ( ! this.props.previewMarkup ) {
+			this.loadPreviewWithProps();
+		}
+		// If the customizations have been removed, restore the original markup
+		if ( this.props.previewMarkup && this.props.customizations && this.props.previewMarkup === prevProps.previewMarkup && prevProps.customizations ) {
+			if ( Object.keys( this.props.customizations ).length === 0 && Object.keys( prevProps.customizations ).length > 0 ) {
+				debug( 'restoring original markup' );
+				// Force the initial markup to be restored because the DOM may have been
+				// changed, and simply not applying customizations will not restore it.
+				this.loadPreviewWithProps();
+			}
+		}
+		// Apply customizations
+		if ( this.props.customizations && this.previewDocument ) {
+			debug( 'updating preview with customizations', this.props.customizations );
+			updatePreviewWithChanges( this.previewDocument, this.props.customizations );
+		}
+	},
+
+	loadPreviewWithProps( props ) {
+		if ( ! props ) {
+			props = this.props;
+		}
```

Comment:
`loadPreviewWithProps` is never called with `props`. I take it it's a leftover detail?